external_apps/common/external_common.go

```diff
@@ -0,0 +1,62 @@
+package common
+
+import (
+	"github.com/Oneledger/protocol/action"
+	"github.com/Oneledger/protocol/data"
+	"github.com/Oneledger/protocol/data/keys"
+	"github.com/Oneledger/protocol/data/transactions"
+	"github.com/Oneledger/protocol/log"
+	"github.com/Oneledger/protocol/storage"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+type ExtTx struct {
+	Tx  action.Tx
+	Msg action.Msg
+}
+
+type ExtServiceMap map[string]interface{}
+
+func NewExtServiceMap() ExtServiceMap {
+	return ExtServiceMap{}
+}
+
+type ExtAppData struct {
+	// we need txs, data stores, services, block functions
+	ChainState *storage.ChainState
+	ExtTxs []ExtTx
+	ExtStores	map[string]data.ExtStore
```

Comment:
I think if you hold alt+ctrl+shift+f it should format your files correctly

Reply:
yeah, forgot that